safetensors/src/tensor.rs

```diff
@@ -385,11 +385,10 @@ impl<'data> SafeTensors<'data> {
             return Err(SafeTensorError::InvalidHeaderLength);
         };
         let string = core::str::from_utf8(header_bytes).map_err(SafeTensorError::InvalidHeader)?;
-        // Assert the string starts with {
-        // NOTE: Add when we move to 0.4.0
-        // if !string.starts_with('{') {
-        //     return Err(SafeTensorError::InvalidHeaderStart);
-        // }
+        // Assert the string starts with `{`
+        if !string.starts_with('{') {
```

Comment:
Is it worth potentially breaking checkpoints over this (not sure if third-party writers made JSON with some initial space)? Maybe it's best to emit a warning for one release, warning that it will be enforced in 0.8 or 0.9?